Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/autorefinement.h

```diff
@@ -602,9 +680,9 @@ void generate_subtriangles(std::size_t ti,
               {
                 CGAL_AUTOREF_COUNTER_INSTRUCTION(counter.timer6.start();)
                 typename EK::Point_3 pt = typename EK::Construct_planes_intersection_point_3()(
-                  triangles[in_triangle_ids[i]][0], triangles[in_triangle_ids[i]][1],triangles[in_triangle_ids[i]][2],
-                  triangles[in_triangle_ids[j]][0], triangles[in_triangle_ids[j]][1],triangles[in_triangle_ids[j]][2],
-                  triangles[ti][0], triangles[ti][1],triangles[ti][2]);
+                  all_triangle_data[in_triangle_ids[i]].points[0], all_triangle_data[in_triangle_ids[i]].points[1],all_triangle_data[in_triangle_ids[i]].points[2],
```

Comment:
With VC2022 and the call `test_autorefinement.exe data-autoref/four_cubes.off 12 94 184 1 78 0 106 352`   I get a segfault in the parallel execution.   My guess (only based on gut feeling) is  that while a thread is in this functor another thread executes the exact construction of one of the input points

Reply:
I can produce an error locally on the same case (need several runs). Thanks for the pointer. Investigating...